src/emake/core/settings.py

```diff
@@ -6,20 +6,31 @@
     import tomli as tomllib
 
 from benedict import benedict
-from typing import Dict, List, Optional
+from typing import Union, Dict, Any
+
+
+def __get_opts(config: Path) -> Union[Dict[str, Any], Dict]:
```

Comment:
A single underscore to indicate a "private" method is fine (and is preferred over two).